sdk/include/compartment.h

```diff
@@ -49,6 +49,113 @@
 		ret;                                                                   \
 	})
 
+/**
+ * Helper macro, used by `STATIC_SEALING_TYPE`.  Do not use this directly, it
+ * exists to avoid error-prone copying and pasting of the mangled name for a
+ * static sealing type.
+ */
+#define CHERIOT_EMIT_STATIC_SEALING_TYPE(name)                                 \
+	({                                                                         \
+		SKey ret; /* NOLINT(bugprone-macro-parentheses) */                     \
+		__asm(                                                                 \
+		  ".ifndef __import." name "\n"                                        \
+		  "  .type     __import." name ",@object\n"                            \
+		  "  .section  .compartment_imports." name ",\"awG\",@progbits," name  \
+		  ",comdat\n"                                                          \
+		  "  .globl    __import." name "\n"                                    \
+		  "  .p2align  3\n"                                                    \
+		  "__import." name ":\n"                                               \
+		  "  .word __export." name "\n"                                        \
+		  "  .word 0\n"                                                        \
+		  " .previous\n"                                                       \
+		  ".endif\n"                                                           \
+		  ".ifndef __export." name "\n"                                        \
+		  "  .type     __export." name ",@object\n"                            \
+		  "  .section  .compartment_exports." name ",\"awG\",@progbits," name  \
+		  ",comdat\n"                                                          \
+		  "  .globl    __export." name "\n"                                    \
+		  "  .p2align  2\n"                                                    \
+		  "__export." name ":\n"                                               \
+		  "  .half 0\n" /* function start and stack size initialised to 0 */   \
+		  "  .byte 0\n"                                                        \
+		  "  .byte 0b100000\n" /* Set the flag that indicates that this is a   \
+		                          sealing key. */                              \
```

Comment:
GitHub has seemingly eaten my earlier suggestion that this be brought in by name via `"i"(SealingTypeEntry)`